### PR TITLE
linuxmain-generator 0.2.0 (new formula)

### PR DESCRIPTION
**Pull request by valeriomazzeo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/linuxmain-generator.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class LinuxmainGenerator < Formula
+  desc "A shell command to automatically add code to Swift Package Manager projects to run unit tests on Linux."
```
Comment (bfontaine) [inline Formula/linuxmain-generator.rb]: Please shorten this; the description should fit in 80 characters.

File: Formula/linuxmain-generator.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  homepage "https://github.com/valeriomazzeo/linuxmain-generator"
+  url "https://github.com/valeriomazzeo/linuxmain-generator/archive/0.2.0.tar.gz"
+  sha256 "84a7a29ef8680426d52591631610b683c853910c49532ad17d7101bd35ed207b"
+
+  def install
+    system "unset CC; swift build -c release"
```
Comment (bfontaine) [inline Formula/linuxmain-generator.rb]: Why do we need to unset `CC` here? Also, please break it into something like:
```rb
ENV["CC"] = ""
system "swift", "build", "-c", "release"
```

If it only builds a binary you can even do `system "swift", "build", "-c", bin.to_s` and remove the `bin.install` below.

File: Formula/linuxmain-generator.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    bin.install ".build/release/linuxmain-generator"
+  end
+  
+  test do
+    system "#{bin}/linuxmain-generator", "--help"
```
Comment (bfontaine) [inline Formula/linuxmain-generator.rb]: Can’t we write a test that does something more substantial than `--help`?

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/linuxmain-generator.rb (hunk 3 above)
Comment (valeriomazzeo) [inline Formula/linuxmain-generator.rb]: not really, it would an example swift project to be shipped with it

Comment (bfontaine) [inline Formula/linuxmain-generator.rb]: Is it hard to create a dummy swift project from scratch?

Comment (valeriomazzeo) [inline Formula/linuxmain-generator.rb]: problem is that it needs the project to be in the same folder: `bin` in this specific case

File: Formula/linuxmain-generator.rb (hunk 2 above)
Comment (valeriomazzeo) [inline Formula/linuxmain-generator.rb]: >If it only builds a binary you can even do system "swift", "build", "-c", bin.to_s
How does it know where to get the binary from? 
As you can see it goes into `.build/release/linuxmain-generator`

**Comment by valeriomazzeo** [comment]
@bfontaine seems something is wrong with your CI as it's not building anymore, it stops at git checkout

**Comment by fxcoudert** [comment]
Can you rebase this into a single commit, clean PR? Or open another one against current master. Thanks.

**Comment by valeriomazzeo** [comment]
sure

File: Formula/linuxmain-generator.rb (hunk 3 above)
Comment (bfontaine) [inline Formula/linuxmain-generator.rb]: So how are users supposed to use a Homebrew-installed `linuxmain-generator`?

**Pull request closed without merging**